### DISCUSSION TITLE
Trend and TrendConstant

### DISCUSSION THREAD

**Discussion by kennethshsu** [casact/chainladder-python#512] [Ideas]
I wanted to start a new thread and propose that we refactor how we use trends in the package.

I believe we should have two separate functions:

First, `chainladder.Trend()`:
- This should be an estimator, but I haven't really thought out how/what models to use. Some ideas:

1. (1+T*i): linearly, is this even useful?
2. (1+i)^T: multiplicatively (default)

Basically, we want to minimize an error function and return the factor `.i_`. But what error function to use? MSE?

Second, `chainladder.TrendConstant()`:
This function should take in a few parameters:
- (array of) `[start date, end date]`
- (array of) `factors`
- `how`? to match the `chainladder.Trend()`, so linearly, and multiplicatively (default)?
- `trend_to`: date to trend to
And returns:
- `trend_factors`, the array of factors that will be applied to the `origin`s

Thoughts?

**Comment by jbogaardt** [comment]
I like your thought process on this one.
